### DISCUSSION TITLE
React native

### DISCUSSION THREAD

**Discussion by avrcoelho** [vitest-dev/vitest#1848] [Q&A]
Hi folks! How are you?

Is React native not supported?

**Comment by sheremet-va** [comment]
It's not supported

**Reply by avrcoelho** [reply]
Thanks for the reply

**Reply by bryanprimus** [reply]
is it still not supported?

**Reply by sheremet-va** [reply]
React Native is supported via [vitest-react-native](https://github.com/sheremet-va/vitest-react-native) package.

**Comment by kraenhansen** [comment]
https://github.com/vitest-dev/vitest/discussions/1848#discussioncomment-6057540

> React Native is supported via [vitest-react-native](https://github.com/sheremet-va/vitest-react-native) package.

I'm personally interested in running vitest on React Native, where the runtime isn't mocked but the test are executed on the actual device / simulator instead of running in a Node.js process. The `vitest-react-native` package is simply mocking the RN runtime APIs. I'm looking for something like the browser-mode but even more generalized: I want to run the vitest test runner in a mode where it's not depending on any runtime APIs outside of the language spec.
